frontend/packages/topology/src/data-transforms/DataModelProvider.tsx

```diff
@@ -19,30 +17,72 @@ interface DataModelProviderProps {
   children?: React.ReactNode;
 }
 
-export const getNamespacedDynamicModelFactories = (
+export const extractDynamicResources = (
   factories: LoadedExtension<DynamicTopologyDataModelFactory>[],
-) =>
-  factories.map(({ properties, ...ext }) => {
-    return {
-      ...ext,
-      properties: {
-        ...properties,
-        resources: (namespace: string) =>
-          Object.assign(
-            {},
-            ...Object.entries(properties.resources).map(([k, v]) => {
-              const kind = v?.model?.version
-                ? referenceForExtensionModel(v.model)
-                : v?.model
-                ? referenceForModel(modelForGroupKind(v.model?.group, v.model?.kind))
-                : v?.opts?.kind;
-
-              return { [k]: { namespace, kind, ...v?.opts } };
-            }),
-          ),
-      },
-    };
+): [
+  CodeRef<WatchResourcesHook>[],
+  LoadedExtension<Omit<DynamicTopologyDataModelFactory, 'resources'>>[],
+] => {
+  const dynamicResources = [];
+  const otherProperties = factories.map(({ properties, ...ext }) => {
+    const { resources, ...other } = properties;
+    if (resources) {
+      dynamicResources.push(resources);
+    }
+    return { properties: other, ...ext };
   });
+  return [dynamicResources, otherProperties];
+};
+
+const WatchResourcesHookResolver: React.FC<{
+  namespace: string;
+  useResources: WatchResourcesHook;
+  onResolve: (resources: WatchK8sResources<any>) => void;
+}> = ({ namespace, useResources, onResolve }) => {
+  const resources = useResources(namespace);
+  React.useEffect(() => {
+    onResolve(resources);
+  }, [onResolve, resources]);
+  return null;
+};
+
+const useFetchedResourceHooks = (
+  dynamicWatchResources: CodeRef<WatchResourcesHook>[],
+): WatchResourcesHook[] => {
+  const [fetchedData, setFetchedData] = React.useState<WatchResourcesHook[]>([]);
+  React.useEffect(() => {
+    Promise.all(dynamicWatchResources.map((resource) => resource()))
+      .then(setFetchedData)
+      .catch(() => setFetchedData([]));
+  }, [dynamicWatchResources]);
+  return fetchedData;
+};
+
+export const DynamicWatchResourcesLoader: React.FC<{
+  namespace: string;
+  dynamicWatchResources: CodeRef<WatchResourcesHook>[];
+  children: (props: WatchK8sResources<any>) => React.ReactNode;
+}> = ({ namespace, dynamicWatchResources, children }) => {
+  const [watchResources, setWatchResources] = React.useState<WatchK8sResources<any>>({});
+  const fetchedResourceHooks = useFetchedResourceHooks(dynamicWatchResources);
```

Comment:
you should be resolving the resources coming directly from the dynamic extensions. I dont think we need this